### PR TITLE
schrodinger server

### PR DESCRIPTION
**Pull request by ZontaNicola**
add server.py with server-side functionality for cleanup, aromatize, dearomatize and CIP stereochemistry labels

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+from flask_cors import CORS, cross_origin
+from  rdkit import Chem
+from  rdkit.Chem import rdDepictor
+from functools import partial
+
+
+app = Flask(__name__)
+CORS(app)
+
+sys.path.insert(0, "/software/lib/Darwin-x86_64/rdkit-Release_2018_03_2/lib/python3.6/site-packages/")
```
Comment (wcjordan) [inline server.py:21]: That's interesting that you're able to import rdkit above before you insert this.  Any idea why that works?

File: server.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+
+
+@app.route('/info', methods=['GET'])
+def info():
+   info = {'imago_version':'no_imago'}
+   return jsonify(info)
+
+
+
```
Comment (wcjordan) [inline server.py:46]: nit there are some cases of extra lines between functions.  I'll just comment here and avoid pointing them all out as this is still an early POC.

File: server.py [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+    return jsonify(json)
+
+
+@app.route('/clean', methods=['POST'])
+def cleanStructure():
+    """
+    Perform clean up of a structure
+    :rtype: json
+    """
+    return runOnMolecule(cleanUp)
```
Comment (wcjordan) [inline server.py:78]: run on molecule is very clever & nice for readability.  You may even be able to make it into a function decorator so the code looks like
```
@app.route('/clean', methods=['POST'])
@runOnMolecule
def cleanStructure(mol):
    """
    Perform clean up of a structure
    :rtype: json
    """
    return cleanUp(mol)
```

File: src/script/api.js [hunk 4]
```diff
@@ -75,13 +75,24 @@ function api(base, defaultOptions) {
...
+			return info.then(() => request(method, url, JSON.stringify(body), {
+				'Content-Type': 'application/json'
+			}));
+		};
+	}
+
 	return Object.assign(info, {
 		convert: indigoCall('POST', 'indigo/convert'),
 		layout: indigoCall('POST', 'indigo/layout'),
-		clean: indigoCall('POST', 'indigo/clean'),
```
Comment (wcjordan) [inline src/script/api.js:81]: if we're stripping out indigo entirely we may want to remove the indigoCall function as well.

File: src/script/index.js [hunk 5]
```diff
@@ -96,7 +96,8 @@ window.onload = function () {
 
 const buildInfo = {
 	version: '__VERSION__',
-	apiPath: '__API_PATH__',
+	apiPath: 'http://127.0.0.1:5000/',
```
Comment (wcjordan) [inline src/script/index.js:99]: How does this get used?  We probably want a TODO to wire this up when we move the backend to the bluebird-chem repo (currently bluebird-standardizer)

File: src/script/ui/app/toolbar.jsx [hunk 6]
```diff
@@ -62,15 +62,15 @@ const mainmenu = [
 	{
 		id: 'process',
 		menu: [
-			'layout',
+			// 'layout',
```
Comment (wcjordan) [inline src/script/ui/app/toolbar.jsx:65]: do we plan to re-enable these options later or are they un-needed?

File: src/script/ui/state/server/index.js [hunk 7]
```diff
@@ -127,6 +142,50 @@ export function serverTransform(method, data, struct) {
...
 }
 
+function deleteAllSGroupsWithName(restruct, action, fieldName) {
+	restruct.molecule.sgroups.forEach((sg, id) => {
+		if (sg.data.fieldName === fieldName)
+			action.mergeWith(fromSgroupDeletion(restruct, id));
+	});
+}
+
+export function calculateCip(result) {
```
Comment (wcjordan) [inline src/script/ui/state/server/index.js:152]: could you add a comment / documentation on this method and deleteAllSGroupsWithName detailing what the methods do & how they're used (for this exported one).

**Review by wcjordan — APPROVED** [review]
Looks good, sorry for me delays taking a look.

**Comment by ZontaNicola** [comment]
I don't think this is needed anymore. Closing the PR

**Pull request closed without merging**